pkg/sql/ttl/ttljob/ttljob.go

```diff
@@ -214,8 +214,15 @@ func (t rowLevelTTLResumer) Resume(ctx context.Context, execCtx interface{}) err
 		if err != nil {
 			return err
 		}
-		if knobs.RequireMultipleSpanPartitions && len(spanPartitions) == 0 {
-			return errors.New("multiple span partitions required")
+		expectedNumSpanPartitions := knobs.ExpectedNumSpanParitions
+		if expectedNumSpanPartitions != 0 {
+			actualNumSpanPartitions := len(spanPartitions)
+			if expectedNumSpanPartitions != actualNumSpanPartitions {
+				return errors.Newf(
```

Comment:
should we make this `AssertionFailedf` which fails a bit more loudly?

Reply:
Yeah good idea